components/ws-manager/pkg/manager/create.go

```diff
@@ -911,6 +911,14 @@ func (m *Manager) newStartWorkspaceContext(ctx context.Context, req *api.StartWo
 		workspaceClassLabel:    clsName,
 	}
 
+	if req.Metadata.Project != "" {
```

Comment:
nit: why not to just add it above into labels map as: 
`wsk8s.ProjectLabel: req.Metadata.Project` ?
If it is empty, it will be an empty label. Which I think is fine.